### ISSUE TITLE
FATAL ERROR: Symbol "ccalllib_libhsa-runtime64445"not found on AMDGPU

### ISSUE THREAD

**Issue by 0x0f0f0f** [JuliaGPU/AMDGPU.jl#73]
Void Linux, glibc. Built `libhsakmt.so` and `libhsa-runtime64.so` manually.

```
(@v1.5) pkg> build AMDGPU
   Building AMDGPU → `~/.julia/packages/AMDGPU/lrlUy/deps/build.log`

(@v1.5) pkg> ^C

julia> using AMDGPU
FATAL ERROR: Symbol "ccalllib_libhsa-runtime64445"not found
signal (6): Aborted
in expression starting at REPL[2]:1
raise at /builddir/glibc-2.30/signal/../sysdeps/unix/sysv/linux/raise.c:51
abort at /builddir/glibc-2.30/stdlib/abort.c:79
unknown function (ip: 0x7f0b696b417a)
unknown function (ip: 0x7f0b697b9a31)
unknown function (ip: 0x7f0b697b9712)
unknown function (ip: 0x7f0b697ba5f5)
unknown function (ip: 0x7f0b697bb458)
unknown function (ip: 0x7f0b6973c28c)
unknown function (ip: 0x7f0b696ee04f)
unknown function (ip: 0x7f0b69704a05)
unknown function (ip: 0x7f0b69705cc8)
unknown function (ip: 0x7f0b69706051)
unknown function (ip: 0x7f0b69709669)
unknown function (ip: 0x7f0b6970fa2f)
unknown function (ip: 0x7f0b6970fb7d)
unknown function (ip: 0x7f0b697103a9)
unknown function (ip: 0x7f0b697ba02d)
unknown function (ip: 0x7f0b697bb458)
unknown function (ip: 0x7f0b6973c28c)
jl_apply_generic at /usr/bin/../lib/libjulia.so.1 (unknown line)
unknown function (ip: 0x7f0b697709a9)
jl_init_restored_modules at /usr/bin/../lib/libjulia.so.1 (unknown line)
unknown function (ip: 0x7f0b5a3d3fb3)
unknown function (ip: 0x7f0b5a4d83cf)
unknown function (ip: 0x7f0b5a60d63d)
unknown function (ip: 0x7f0b5a558c1e)
unknown function (ip: 0x7f0b5a5597bc)
unknown function (ip: 0x7f0b697705fd)
unknown function (ip: 0x7f0b69771b32)
unknown function (ip: 0x7f0b69771c98)
jl_toplevel_eval_in at /usr/bin/../lib/libjulia.so.1 (unknown line)
unknown function (ip: 0x7f0b5a2d9c81)
unknown function (ip: 0x7f0b5a03162e)
unknown function (ip: 0x7f0b5a032270)
unknown function (ip: 0x7f0b5a03239a)
unknown function (ip: 0x7f0b5a048ae7)
run_repl at /builddir/julia-1.5.2/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:288
unknown function (ip: 0x7f0b5a1dbfdf)
unknown function (ip: 0x7f0b5a1dc0b8)
unknown function (ip: 0x7f0b697498c0)
jl_f__apply_latest at /usr/bin/../lib/libjulia.so.1 (unknown line)
unknown function (ip: 0x7f0b5a2054a7)
unknown function (ip: 0x7f0b5a210b07)
unknown function (ip: 0x7f0b5a2123ae)
unknown function (ip: 0x7f0b5a212505)
unknown function (ip: 0x5600ebef4755)
unknown function (ip: 0x5600ebef4332)
__libc_start_main at /builddir/glibc-2.30/csu/../csu/libc-start.c:308
unknown function (ip: 0x5600ebef43d9)
Allocations: 4838761 (Pool: 4836877; Big: 1884); GC: 6
Aborted
```

**Comment by 0x0f0f0f** [comment]
Is this because ROCm uses its own fork of llvm?

**Comment by 0x0f0f0f** [comment]
```
cat ~/.julia/packages/AMDGPU/lrlUy/deps/build.log
[ Info: libhsa-runtime64.so: true
[ Info: Found useable ld.lld at /usr/bin/ld.lld
paths = ["/opt/rocm/hsa/lib"]
Could not find library 'rocblas'.
Could not find library 'rocsparse'.
Could not find library 'rocalution'.
Could not find library 'rocfft'.
Could not find library 'rocrand'.
Could not find library 'MIOpen'.
```

will build and let you know

**Comment by jpsamaroo** [comment]
> Is this because ROCm uses its own fork of llvm?

Possibly, but I'm not sure of that from just this stacktrace. If you can build julia-debug (`make julia-debug` on <=1.5 I believe), you should be able to see the Julia functions in all those `unknown function` entries.

**Comment by jpsamaroo** [comment]
Closed by #105
